Add Page for Data Module (#613)
PennyLaneAI/qml: datasets_intro.rst

.. raw:: html

    <style>
        h1 {
            text-align: center;
        }
    </style>

    <link href="https://cdnjs.cloudflare.com/ajax/libs/mdbootstrap/4.5.14/css/mdb.min.css" rel="stylesheet">

What is a Quantum Dataset?
=====================

.. meta::
   :property="og:description": Every second spent in training parameterized circuits is a second spent not doing greater things in quantum computing.
   :property="og:image": https://pennylane.ai/qml/_static/datasets.png

A quantum dataset is a collection of **quantum data** obtained from various quantum systems that describes it and its evolution.

> Quantum Data
> ~~~~~~~~~~~~~~

Reviewer: ```suggestion
Examples of quantum data
~~~~~~~~~~~~~~~~~~~~~~~~
```

Just trying to make this clearer; when I first read this section, I thought it was describing _categories_ of quantum data, but on-rereading it is more _examples_ of quantum date.

.. image:: /_static/datasets/qdata-illustration.png
    :align: right
    :width: 30%
    :target: javascript:void(0);

> In a more general sense, we consider the quantum data to encompass the following things:

Reviewer: ```suggestion
Examples of quantum data include the following:
```

#. **Hamiltonian** of the system and any other auxiliary observables for other relevant properties of interest.
#. **Quantum state** of interest for the system such as the ground state, and an efficient **state-preparation circuit** for it.

> #. Any useful **unitary transformations** of the system such as the Clifford operator required for tapering.

Reviewer: ```suggestion
#. Any useful **unitary transformations** of the system such as the Clifford operator required for qubit tapering.
```

A good opportunity to link to the qubit tapering demo (using `:doc:`)

#. **Measurement or projection operators** for the systems and any resulting distributions and expectation values.
#. Control parameters for **system evolution** and data related to **noise descriptions** such as the Kraus operators for channels.

Motivating Quantum Datasets
~~~~~~~~~~~~~~~~~~~~~~~~~~~~

.. image:: /_static/datasets/qdata-motivation.png
    :align: left
    :width: 25%
    :target: javascript:void(0);

#. **A lesson from the success of machine learning:** Availability of multi-scale, high-quality accessible datasets for training and benchmarking acts as a catalysis of both developing new algorithms, and improving older ones. 
#. **Challenge of achieving quantum advantage:** Learning from quantum data might be more intuitive for quantum computers than classical ones, leading to an ideal candidate for quantum computational advantage.

> #. **Never-ending quest for useful research:** Readily available data reduces the work required for cross-collaboration among different disciplines, ultimately fostering better advancements in current algorithmic techniques. 

Reviewer: Nice!

Exploring quantum datasets using PennyLane
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> PennyLane is an **open-source software framework built around the concept of quantum differentiable programming**. It seamlessly integrates quantum simulators and hardware with machine-learning libraries, giving users the power to use the provided datasets to develop novel algorithms and benchmark them for problems in applied quantum computation, such as the simulation of physical systems. We also provide the **``data`` module** in PennyLane to access the hosted dataset readily and store and manipulate them locally.

Reviewer: ```suggestion
PennyLane is an **open-source software framework built around the concept of quantum differentiable programming**, seamlessly integrating quantum simulators and hardware with machine-learning libraries.

PennyLane datasets give you the power to use the provided datasets to develop novel algorithms, and benchmark them against problems in applied quantum computation, such as the simulation of physical systems.

For more information on accessing hosted data from PennyLane, please see the :doc:`datasets <introduction/quantum_datasets>` documentation.
```

> To find out more, visit the `PennyLane Documentation <https://pennylane.readthedocs.io>`_, or check out the `Accessing Datasets` section of :doc:`quantum datasets <datasets>`.

Reviewer: ```suggestion
```

.. raw:: html

    <style>
        h1 {
            text-align: center;
        }
    </style>

    <link href="https://cdnjs.cloudflare.com/ajax/libs/mdbootstrap/4.5.14/css/mdb.min.css" rel="stylesheet">

What is a Quantum Dataset?
==========================

.. meta::
   :property="og:description": Browse our collection of quantum datasets and import them into PennyLane directly from your code.
   :property="og:image": https://pennylane.ai/qml/_static/datasets.png

A quantum dataset is a collection of **quantum data** that describes quantum systems and their evolution.

> In the context of quantum programming, this can be realized as the input and output of a quantum function that describes a quantum system.

Reviewer 2: I feel like the definition of "quantum data" is not direct enough here. If quantum data is information that describes quantum systems and their evolution, I would like that to be said explicitly. But I haven't been able to think of a way to make this flow nicely and keep the the statement that "a quantum dataset is a collection of quantum data"

Reviewer: I have to admit I keep reading the line "a quantum dataset is a collection of quantum data" now, and I'm not sure it means anything haha. It sounds tautological.

Examples of quantum data
~~~~~~~~~~~~~~~~~~~~~~~~

.. image:: /_static/datasets/qdata-illustration.png
    :align: right
    :width: 30%
    :target: javascript:void(0);

Examples of quantum data include the following:

#. **Hamiltonian** of the system and any other auxiliary observables for other relevant properties of interest.
#. **Quantum state** of interest for the system such as the ground state, and an efficient **state-preparation circuit** for it.
#. Any useful **unitary transformations** of the system such as the Clifford operator required for :doc:`qubit tapering <demos/tutorial_qubit_tapering.html>`.
#. **Measurement or projection operators** for the systems and any resulting distributions and expectation values that can be extracted for the system.

> #. Control parameters for **system evolution** and data related to **noise descriptions** such as the Kraus operators for channels.

Author: No, but we could have them in the future with some noisy experiments (either on a simulator or hardware).

#. **A lesson from the success of machine learning:** Availability of multi-scale, high-quality accessible datasets for training and benchmarking acts as a catalysis of both developing new algorithms, and improving older ones. 
#. **Challenge of achieving quantum advantage:** Learning from quantum data might be more intuitive for quantum computers than classical ones, leading to an ideal candidate for quantum computational advantage.
#. **Never-ending quest for useful research:** Readily available data reduces the work required for cross-collaboration among different disciplines, ultimately fostering better advancements in current algorithmic techniques. 

Exploring quantum datasets using PennyLane
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

PennyLane is an **open-source software framework built around the concept of quantum differentiable programming**, seamlessly integrating quantum simulators and hardware with machine-learning libraries.

PennyLane datasets give you the power to use the provided datasets to develop novel algorithms, and benchmark them against problems in applied quantum computation, such as the simulation of physical systems.

> For more information on accessing hosted data from PennyLane, please see the :doc:`datasets <introduction/quantum_datasets>` documentation.

Reviewer 3: is this supposed to be linking to the [PennyLane doc](https://docs.pennylane.ai/en/latest/introduction/data.html)?
